server/src-lib/Data/Aeson/Ordered.hs

```diff
@@ -0,0 +1,221 @@
+{-# LANGUAGE CPP                #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+
+-- | A version of aeson that parses with key order preserved.
```

Comment:
Yes, this is basically a port of some aeson version which uses ordered map instead of hashmap which Chris had. I think we can pick out the version from some fpcomplete repo. This is used for maintaining the order of the query during serialization. 

I am not sure how much we should worry about this atm.

Reply:
Although, if we can think of some other way of serialization then that might be more than better. Because ordered map is also not the best datastructure here as it is ordered "according to time of insert" rather than order according to some generic value. It works for us because the time property holds for us currently.

Reply:
Thanks, that makes sense, and agree probably not a priority atm. Would be great to document this, esp your second comment there; I don't think it occurred to me that that's how it worked and that does sound fragile.